### PR TITLE
power9 makefile. dgemm based on power8 kernel with following changes …

### PR DESCRIPTION
**Pull request by quickwritereader**
…: 32x unrolled 16x4 kernel and 8x4 kernel using (lxv stxv butterfly rank1 update). improvement from 17 to 22-23gflops. dtrmm cases were added into dgemm itself.
Though it is below desired 90% peak level and around 78-80% we decided to publish as it is still improved version above current kernel.

### CODE REVIEW

**Comment by quickwritereader** [comment]
#2016

File: kernel/power/dgemm_kernel_power9.S [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    stxv    v29,  432(SP)
+    stxv    v30,  448(SP)
+    stxv    v31,  464(SP)
+
+
+	stfd	f1,  ALPHA_SP
+	stw	r0,  FZERO 
+
+	slwi	LDC, LDC, BASE_SHIFT
+
```
Comment (yuriy-yarosh) [inline kernel/power/dgemm_kernel_power9.S:156]: lwsync before a massive store might help with op ordering a bit.

File: kernel/power/dgemm_kernel_power9.S [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+#ifndef NEEDPARAM
+
+	PROLOGUE
+	PROFCODE
+
+	addi	SP, SP, -STACKSIZE
+	li	r0, 0
+
```
Comment (yuriy-yarosh) [inline kernel/power/dgemm_kernel_power9.S:93]: It should be a good idea to add a cache hint with dcbtstds over here, with TH=0b01010 and the respective units count specified.

File: kernel/power/dgemm_kernel_power9.S [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+	li	o48, 48
+
+
+	lxvdsx	alpha_r, 0, T1
+
+#include "dgemm_logic_power9.S"
+
+.L999:
+	addi	r3, 0, 0
+
```
Comment (yuriy-yarosh) [inline kernel/power/dgemm_kernel_power9.S:188]: Same goes for loads dcbtds, and TH=0b01010 would be great, but I think it worth also marking this load as Transient.

File: kernel/power/dgemm_logic_power9.S [hunk 4]
```diff
@@ -0,0 +1,1981 @@
...
+	li		T1,	128
+	li		T2,	256
+ 
+	mr		AO,	A
+	mr		CO,	C
+	slwi		T3,	LDC	,	2
+	add		C,	C,	T3
+
+ 
+	dcbt		A,	T1
```
Comment (yuriy-yarosh) [inline kernel/power/dgemm_logic_power9.S:50]: It worth using dcbtds with TH=0b01010 and Transient flag over here.
It's worth using extended mnemonics for dcbt - it makes code a bit more readable.

Comment (quickwritereader) [reply kernel/power/dgemm_logic_power9.S:50]: @yuriy-yarosh Thanks for the valuable advice Frankly, While patching I went on ignoring and not adding prefetch instructions on both sgemm/dgemm to focus on correctness and initial architectural problems.  I will definitely consider adding them back in a  proper and readable way to improve performance further.

**Pull request merged**